File: react/src/commentbox/CommentForm.js
import React, {Component} from 'react';

class CommentForm extends Component {

    constructor() {
        super();
        this._handleSubmit = this._handleSubmit.bind(this);
        this._updateAuthor = this._updateAuthor.bind(this);
        this._updateComment = this._updateComment.bind(this);

        this.state = {
            author: "",
            comment: ""
        }

    }

    render() {
        return (
            <form className="comment-form" onSubmit={this._handleSubmit}>
                <input placeholder ="author" value={this.state.author} onChange={this._updateAuthor}/>
                <input placeholder="comment" value={this.state.comment} onChange={this._updateComment}/>
                <button type="submit">Post Comment</button>
            </form>
        );
    }

    _handleSubmit(event) {
        event.preventDefault();
        this.props.add(this.state.author, this.state.comment);
        this.setState({ author: "", comment: ""});
    }

    _updateAuthor(event) {
        this.setState({
            author : event.target.value
        })
    }

    _updateComment(event) {
        this.setState({
            comment : event.target.value
        })
    }
}

export default CommentForm;

File: src/main/scala/samples/r4s/CommentForm.scala
package samples.r4s

import com.github.ahnfelt.react4s._
import samples.r4s.CommentForm.{AddComment, CommentFormMsg}

case class CommentForm() extends Component[CommentFormMsg] {

  val author  = State("")
  val comment = State("")

  override def render(get: Get): Element = {
    E.div(
      E.h3(Text("CommentForm")),
      E.form(
        E.input(A.onChangeText(author.set), A.value(get(author))),
        E.input(A.onChangeText(comment.set), A.value(get(comment))),
        E.button(Text("Post Comment#"))
      ),
      A.onSubmit { e =>
        e.preventDefault()
        emit(AddComment(get(author), get(comment)))
        author.set("")
        comment.set("")
      }
    )
  }

}

object CommentForm {
  sealed trait CommentFormMsg
  case class AddComment(author: String, comment: String) extends CommentFormMsg
}
